Plugins/GitHub3/GitHub3Plugin.cs

```diff
@@ -146,6 +158,80 @@ public override void Register(IGitUICommands gitUiCommands)
             {
                 GitHub.setOAuth2Token(GitHubLoginInfo.OAuthToken);
             }
+
+            gitUiCommands.PreCommit += GitUiCommands_PreCommit;
+            gitUiCommands.PostCommit += GitUiCommands_PostCommit;
+        }
+
+        private void GitUiCommands_PreCommit(object? sender, GitUIEventArgs e)
+        {
+            if (!_issueCommitMessageHelperEnabled.ValueOrDefault(Settings))
+            {
+                return;
+            }
+
+            if (string.IsNullOrEmpty(GitHubLoginInfo.OAuthToken))
+            {
+                e.GitUICommands.AddCommitTemplate(_noTokenError.Text, () => string.Empty, Icon);
+                return;
+            }
+
+            ThreadHelper.FileAndForget(async () =>
+            {
+                IHostedRemote[] hostedRemotes = GetHostedRemotes().ToArray();
+                if (hostedRemotes.Length == 0)
+                {
+                    return;
+                }
+
+                IReadOnlyList<Issue> issues = _gitHub.GetAssignedIssues();
+
+                if (issues?.All(i => i.Number == 0) ?? true)
+                {
+                    await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                    e.GitUICommands.AddCommitTemplate(_noAssignedIssues.Text, () => string.Empty, Icon);
+                    return;
+                }
+
+                Issue[] recentUserIssues = issues.Where(i => i.Number != 0 && hostedRemotes.Any(r => r.Owner == i.Repository.Owner.Login && r.RemoteRepositoryName == i.Repository.Name))
+                                                            .OrderByDescending(i => i.UpdatedAt)
+                                                            .Take(MaxMostRecentIssuesDisplayed)
+                                                            .ToArray();
+
+                bool multipleRemotes = hostedRemotes.Length > 1;
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                foreach (Issue issue in recentUserIssues)
+                {
+                    string remoteData = multipleRemotes ? $" ({issue.Repository.Owner.Login}/{issue.Repository.Name})" : string.Empty;
+                    string key = $"{issue.Number}: {issue.Title}{remoteData}";
+                    _currentMessages.Add(key);
+                    e.GitUICommands.AddCommitTemplate(key, () => GetIssueDescription(issue), Icon);
+                }
+
+                string GetIssueDescription(Issue issue)
+                    => $@"
+
+Fixes #{issue.Number} : {issue.Title}
+
+{issue.Body}
+
+";
```

Comment:
raw string
(did not count spaces)
```suggestion
                    => $@"""

                               Fixes #{issue.Number} : {issue.Title}

                               {issue.Body}

                               """;
```